### ISSUE TITLE
Fetch upstream / setlinkrev UI

### ISSUE THREAD

**Issue by hennevogel** [openSUSE/open-build-service#11676]
Github introduced a nice UI for syncing your fork with upstream.
 
![Github fetch upstream UI](https://user-images.githubusercontent.com/514785/135314738-0d3c8af2-9aab-4bbd-8cfb-aa7d75c1b236.gif)

We could use something like that for package links. osc has the `setlinkrev` command for this.

**Comment by Yash621** [comment]
@hennevogel can I work on this ?

**Comment by dmarcoux** [comment]
> @hennevogel can I work on this ?

For sure! Let us know if something isn't clear :slightly_smiling_face:

**Comment by Yash621** [comment]
@dmarcoux can you tell in which files I would get the setup for this ?

**Comment by dmarcoux** [comment]
Do you mean setting up a development environment? If so, it's explained in the [contributing guide](https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md#how-to-setup-an-obs-development-environment).

**Comment by adrianschroeter** [comment]
setlinkrev is not exactly the same as merging with upstream (this happens in OBS source server anyway by default, except someone played with setting linkrevisions). It is more like a rebase in git terminlogy.

It would be good if the button makes it clear (eg. saying "freezing sources" when no link revision is specified yet (what is the case for the vast majority of packages).  Otherwise people will set link revisions even though they never wanted that.)

**Comment by hennevogel** [comment]
Of course, wouldn't make much sense to have any button if things are already in sync...
